operator/src/main/java/com/github/streamshub/console/api/v1alpha1/spec/KafkaConnect.java

```diff
@@ -0,0 +1,113 @@
+package com.github.streamshub.console.api.v1alpha1.spec;
+
+import java.util.ArrayList;
+import java.util.List;
+
+import com.fasterxml.jackson.annotation.JsonInclude;
+import com.fasterxml.jackson.annotation.JsonPropertyDescription;
+import com.github.streamshub.console.api.v1alpha1.spec.authentication.Authentication;
+
+import io.fabric8.generator.annotation.Required;
+import io.sundr.builder.annotations.Buildable;
+
+@Buildable(editableEnabled = false)
+@JsonInclude(JsonInclude.Include.NON_NULL)
+public class KafkaConnect {
+
+    @Required
+    @JsonPropertyDescription("""
+            Name of the Kafka Connect cluster. When the console is connected to
+            Kubernetes, a KafkaConnect custom resource may be discovered using
+            this property together with the namespace property.
+            """)
+    private String name;
+
+    @JsonPropertyDescription("""
+            The namespace of the Kafka Connect cluster. When the console is connected to
+            Kubernetes, a Strimzi KafkaConnect custom resource may be discovered using
+            this property together with the name property.
+            """)
+    private String namespace;
+
+    @Required
+    @JsonPropertyDescription("URL of the Kafka Connect REST API endpoint.")
+    private String url;
+
+    @JsonPropertyDescription("""
+            Indicates whether this Kafka Connect cluster is a MirrorMaker 2 deployment.
+            This affects how certain resources are displayed and managed by the console.
+            Defaults to false.
+            """)
+    private Boolean mirrorMaker = Boolean.FALSE;
```

Comment:
Better to let this remain `null` and handle it (as you've already done in the ConfigurationProcessor) as possibly null. Initializing fields in the CR classes has the side effect of making changes to the CR in Kube. Sometimes that might be fine, but usually the operator should only make change to the status section itself.

```suggestion
    private Boolean mirrorMaker;
```